### PR TITLE
[PDS-176064] burning smoke and bullet fire | getSortedColumns now avoids local writes correctly

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- Fix a bug in `getSortedColumns`: when checking our reads against the read-only transaction, we should only check cells for which there were no _local_ writes. This brings this in line with the behaviour of `getRowsColumnRange`.

**Implementation Description (bullets)**:
- When checking our reads against the read-only transaction, we should only check cells for which there were no _local_ writes.
- I wrote two tests that would have previously failed but now pass. (The another-table test is just there because it's something that could accidentally go wrong as well.)

**Testing (What was existing testing like?  What have you done to improve it?)**:
- Added new tests that are evidence of the previous bug.

**Concerns (what feedback would you like?)**:
- Are there any other edge cases I missed out?

**Where should we start reviewing?**: `SerializableTransactionTest`

**Priority (whenever / two weeks / yesterday)**: yesterday 🔥

### CODE REVIEW

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java [hunk 1]
```diff
@@ -1201,6 +1204,101 @@ public void testGetSortedColumnsReadWriteConflictAtEndOfBatch() {
...
+        List<Map.Entry<Cell, byte[]>> sortedColumnValues =
+                Streams.stream(sortedColumns).collect(Collectors.toList());
+        List<Cell> sortedCellsRead =
+                sortedColumnValues.stream().map(Entry::getKey).collect(Collectors.toList());
+        sanityCheckOnSortedCells(rows, sortedCellsRead, cellsWrittenOriginally);
+        assertThat(sortedColumnValues)
+                .contains(
+                        Maps.immutableEntry(cellsWrittenOriginally.get(0), newValue1),
+                        Maps.immutableEntry(cellsWrittenOriginally.get(17), newValue2));
+        t1.commit();
```
Comment (sudiksha27) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java:1233]: nit: given the name of the test - 
```suggestion
        assertThatCode(t1::commit).doesNotThrowAnyException();
```

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java [hunk 2]
```diff
@@ -1201,6 +1204,101 @@ public void testGetSortedColumnsReadWriteConflictAtEndOfBatch() {
...
+                Streams.stream(sortedColumns).collect(Collectors.toList());
+        List<Cell> sortedCellsRead =
+                sortedColumnValues.stream().map(Entry::getKey).collect(Collectors.toList());
+        sanityCheckOnSortedCells(rows, sortedCellsRead, cellsWrittenOriginally);
+        assertThat(sortedColumnValues)
+                .doesNotContain(
+                        Maps.immutableEntry(cellsWrittenOriginally.get(0), newValue1),
+                        Maps.immutableEntry(cellsWrittenOriginally.get(17), newValue2));
+        t1.commit();
+    }
```
Comment (sudiksha27) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java:1263]: same as above.

**Review by sudiksha27 — APPROVED** [review]
Makes sense!

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java [hunk 3]
```diff
@@ -1201,6 +1204,101 @@ public void testGetSortedColumnsReadWriteConflictAtEndOfBatch() {
...
+        List<Map.Entry<Cell, byte[]>> sortedColumnValues =
+                Streams.stream(sortedColumns).collect(Collectors.toList());
+        List<Cell> sortedCellsRead =
+                sortedColumnValues.stream().map(Entry::getKey).collect(Collectors.toList());
+        sanityCheckOnSortedCells(rows, sortedCellsRead, cellsWrittenOriginally);
+        assertThat(sortedColumnValues)
+                .doesNotContain(
+                        Maps.immutableEntry(cellsWrittenOriginally.get(0), newValue1),
+                        Maps.immutableEntry(cellsWrittenOriginally.get(17), newValue2));
+        t1.commit();
```
Comment (jeremyk-91) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java:1262]:
```suggestion
       assertThatCode(t1::commit).doesNotThrowAnyException();
```
following @sudiksha27's comment after this one so I can batch them in the ui
